packages/eslint-plugin/src/rules/prefer-ts-expect-error.ts

```diff
@@ -8,44 +15,58 @@ export default util.createRule<[], MessageIds>({
   meta: {
     type: 'problem',
     docs: {
-      description:
-        'Recommends using `// @ts-expect-error` over `// @ts-ignore`',
+      description: 'Recommends using `@ts-expect-error` over `@ts-ignore`',
       category: 'Best Practices',
       recommended: false,
     },
     fixable: 'code',
     messages: {
       preferExpectErrorComment:
-        'Use "// @ts-expect-error" to ensure an error is actually being suppressed.',
+        'Use "@ts-expect-error" to ensure an error is actually being suppressed.',
     },
     schema: [],
   },
   defaultOptions: [],
   create(context) {
-    const tsIgnoreRegExp = /^\/*\s*@ts-ignore/;
+    const tsIgnoreRegExp = /(^|\n)((\*)*(\/)*\s*(\/\*)*\s*)*\s*@ts-ignore/;
```

Comment:
Not sure if that's a problem, since in that part of a code we already know that we're "inside" the valid comment to my understanding, but thig RegExp currently mathces invalid comments like `/ * @ts-ignore` or `////* @ts-ignore`

Reply:
we might want to tighten up this handling and separate the logic for block and line comments.

These are the regexes used by TS itself:

https://github.com/microsoft/TypeScript/blob/bbf26a07eb2ad8c5e117d6128f01e0a4c31f5687/src/compiler/scanner.ts#L284-L292

Unfortunately they're not exported, so we'll have to copy paste them - but that's fine.
This will ensure we're aligned completely with TS.

Reply:
I had a look at these yesterday, because pushing my changes. The problem is that in Typescript code base, the regex is run on each line of the code which is supplied by scanner.ts

Here is it run on the stripped down version of the comments. This is because the `sourceCode.getAllComments` already removes all comment indicators.

For example - 
`// @ts-ignore` --> ` @ts-ignore`

and 

`/** 
*
* @ts-ignore */ ` will get transformed into 

`*
*
* @ts-ignore`

So we can't use them as-is - we might need to account for missing `/*`, `//` and even `\n`.

I think the correctness is enforced by TS and we shouldn't strive for that. Plus using the same regex leaves us open for loads of rework tasks if the parent regexs change.

IMO we should just look to find instances of @ts-ignore and suggest fixes for it. Correctness should be left to TS

@bradzacher  and @MHMadHamster (Sorry I am unable to tag you)- Your thoughts ?

Reply:
I have added the 2 TS regex after modifications to accommodate changes that we do for `getAllComments`.

The fixers are still separate though because getting the range for `@ts-ignore` had a little different treatment for `blockComment` and `lineComment` and it seemed error prone. More details on the other comment